test/integrations/testTypes.ts

```diff
@@ -44,3 +45,98 @@ export type MockHttpCallsData = {
   httpReq: Record<string, any>;
   httpRes: Partial<AxiosResponse>;
 };
+
+export const arrayOrSingleElement = (schema: ZodTypeAny) => z.union([z.array(schema), schema]);
+export const formElementSchema = arrayOrSingleElement(
+  z.union([z.string(), z.number(), z.boolean()]),
+);
+export const eventResponseSchema = z.union([
+  z.object({
+    error: z.string(),
+    statTags: z
+      .object({
+        destType: z.string().optional(),
+        errorCategory: z.string().optional(),
+        errorType: z.string().optional(),
+        feature: z.string().optional(),
+        implementation: z.string().optional(),
+        module: z.string().optional(),
+      })
+      .optional(),
+    statusCode: z.number().optional(),
+  }),
+  z.object({
+    output: z
+      .object({
+        version: z.string().optional(),
+        type: z.string().optional(),
+        method: z.string().optional(),
+        endpoint: z.string().optional(),
+        headers: z.record(z.string(), z.string()).optional(),
+        params: z.record(z.string(), z.any()).optional(),
+        body: z
+          .object({
+            JSON: z.record(z.string(), z.any()).optional(),
+            XML: z
+              .object({
+                payload: z.string().optional(),
+              })
+              .optional(),
+            JSON_ARRAY: z
+              .object({
+                batch: z.any().optional(),
+              })
+              .optional(),
+            FORM: z.record(z.string(), formElementSchema).optional(),
+            GZIP: z
+              .object({
+                payload: z.string().optional(),
+              })
+              .optional(),
+          })
+          .optional(),
+        files: z.record(z.string(), z.any()).optional(),
+        userId: z.string().optional(),
+      })
+      .optional(),
+    metadata: z.record(z.string(), z.any()).optional(),
+    statusCode: z.number().optional(),
+  }),
+]);
+
+export const batchTransformSchema = z.array(
+  z.object({
+    batchedRequest: arrayOrSingleElement(eventResponseSchema).optional(),
```

Comment:
already `metadata` is captured in eventResponseSchema.output.metadata

Reply:
that is mainly for process response